### PR TITLE
Add skip_tools haplotypecaller_filter

### LINKED ISSUE #876: Make the filtering step in the haplotype_single_sample track skippable
### Description of feature

For some use cases, it would be useful to skip the filtering following Haplotypecaller in the single sample mode:

https://nfcore.slack.com/archives/CGFUX04HZ/p1669759507612189?thread_ts=1665054217.962299&cid=CGFUX04HZ
https://nfcore.slack.com/archives/CGFUX04HZ/p1668784847234849

### PR DESCRIPTION
**Pull request by maxulysse**
close #876

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: workflows/sarek.nf [hunk 1]
```diff
@@ -954,6 +954,7 @@ workflow SAREK {
         // GERMLINE VARIANT CALLING
         BAM_VARIANT_CALLING_GERMLINE_ALL(
             params.tools,
+            params.skip_tools,
```
[the remark this answers is not in the document]
Comment (maxulysse) [reply workflows/sarek.nf:957]: good point, will try that out

File: subworkflows/local/bam_variant_calling_haplotypecaller/main.nf [hunk 2]
```diff
@@ -126,31 +126,34 @@ workflow BAM_VARIANT_CALLING_HAPLOTYPECALLER {
...
-                    sex:meta.sex,
-                    id:meta.sample,
-                    num_intervals:meta.num_intervals,
-                    variantcaller:"haplotypecaller"
-                ],
-                vcf
-            ]
+        if (!(skip_tools && skip_tools.split(',').contains('haplotypecaller_filter'))) {
+
+            VCF_VARIANT_FILTERING_GATK(haplotypecaller_vcf.join(haplotypecaller_tbi),
```
Comment (FriederikeHanssen) [inline subworkflows/local/bam_variant_calling_haplotypecaller/main.nf:131]: should this also skip filtering for joint_germline?

Comment (FriederikeHanssen) [reply subworkflows/local/bam_variant_calling_haplotypecaller/main.nf:131]: fair enough, can we name it more like haplotypecaller_skip_singlefilter ?  or so, I am expecting confusion when mixing `--joint_germline` and `skip_filter`

File: workflows/sarek.nf (hunk 1 above)
[the remark this answers is not in the document]
Comment (maxulysse) [reply workflows/sarek.nf:957]: No failures due to that

**Comment by maxulysse** [comment]
Just need to add the unfiltered vcf to the csv file

File: subworkflows/local/bam_variant_calling_haplotypecaller/main.nf [hunk 3]
```diff
@@ -126,31 +126,34 @@ workflow BAM_VARIANT_CALLING_HAPLOTYPECALLER {
...
+                        fasta,
+                        fasta_fai,
+                        dict,
+                        intervals_bed_combined,
+                        known_sites_indels.concat(known_sites_snps).flatten().unique().collect(),
+                        known_sites_indels_tbi.concat(known_sites_snps_tbi).flatten().unique().collect())
+
+            filtered_vcf = VCF_VARIANT_FILTERING_GATK.out.filtered_vcf.map{ meta, vcf-> [
```
Comment (FriederikeHanssen) [inline subworkflows/local/bam_variant_calling_haplotypecaller/main.nf:139]: For the output of the sw: nono-filtered, if it is skipped? and then use just hc output for annotation?

Comment (maxulysse) [reply subworkflows/local/bam_variant_calling_haplotypecaller/main.nf:139]: yes, that's what I was thinking about when I mentioned the csv file

File: subworkflows/local/bam_variant_calling_haplotypecaller/main.nf (hunk 2 above)
Comment (maxulysse) [reply subworkflows/local/bam_variant_calling_haplotypecaller/main.nf:131]: we can skip filtering in `joint_germline` as well with this `--skip_tools haplotypecaller_filter`, it does make sense to do so

Comment (maxulysse) [reply subworkflows/local/bam_variant_calling_haplotypecaller/main.nf:131]: actually not sure what to skip or not to skip there, prefer to do so in a separate PR

Comment (FriederikeHanssen) [reply subworkflows/local/bam_variant_calling_haplotypecaller/main.nf:131]: I was thinking of the VQSR steps, but good for me to do so in a separate PR

**Comment by FriederikeHanssen** [comment]
Just the changelog missing, i think :)

**Review by FriederikeHanssen — APPROVED** [review]

File: subworkflows/local/bam_variant_calling_haplotypecaller/main.nf [hunk 4]
```diff
@@ -17,12 +17,12 @@ workflow BAM_VARIANT_CALLING_HAPLOTYPECALLER {
...
     known_sites_snps_tbi
     intervals_bed_combined          // channel: [mandatory] intervals/target regions in one file unzipped, no_intervals.bed if no_intervals
-
+    skip_haplotypecaller_filter
 
     main:
 
-    ch_versions = Channel.empty()
-    filtered_vcf = Channel.empty()
+    versions = Channel.empty()
```
Comment (asp8200) [inline subworkflows/local/bam_variant_calling_haplotypecaller/main.nf:24]: Are we in general replacing `ch_versions` with `versions`?

**Pull request merged**